src/client/builder.rs

```diff
@@ -1,29 +1,37 @@
-use super::{connection::*, AuthMethod};
-use crate::{tds::Context, Client, EncryptionLevel};
+use super::AuthMethod;
+use crate::EncryptionLevel;
 use std::collections::HashMap;
 
 #[derive(Clone, Debug)]
 /// A builder for creating a new [`Client`].
 ///
 /// [`Client`]: struct.Client.html
-pub struct ClientBuilder {
-    host: Option<String>,
-    port: Option<u16>,
-    database: Option<String>,
-    #[cfg(windows)]
-    instance_name: Option<String>,
-    encryption: EncryptionLevel,
-    trust_cert: bool,
-    auth: AuthMethod,
+pub struct ClientBuilder
+{
+    pub (crate) host: Option<String>,
+    pub (crate) port: Option<u16>,
+    pub (crate) database: Option<String>,
+    //#[cfg(windows)]
+    pub (crate) instance_name: Option<String>,
+    pub (crate) encryption: EncryptionLevel,
+    pub (crate) trust_cert: bool,
+    pub (crate) auth: AuthMethod,
 }
 
-impl Default for ClientBuilder {
-    fn default() -> Self {
-        Self {
+impl ClientBuilder
+{
+    /// Create a `ClientBuilder` using a connector
+    /// The `connector` must be able to create a TCP connection
+    /// with the remote server. To do this it must implement `AsyncRead`
+    /// and `AsyncWrite` but it is up to the user to ensure that it does
+    /// this via the TCP protocol.
+    pub fn new(
+    ) -> ClientBuilder {
+        ClientBuilder {
             host: None,
             port: None,
             database: None,
-            #[cfg(windows)]
+            //#[cfg(windows)]
```

Comment:
?

Reply:
awks! commented these out to verify if the named-instance stuff could compile